test/test_fips.rb

```diff
@@ -8,4 +8,12 @@ def test_fips_mode_is_reentrant
     OpenSSL.fips_mode = false
   end
 
+  def test_fips_mode_returns_right_boolean
+    if OpenSSL::OPENSSL_FIPS
+      OpenSSL.fips_mode = true
+      OpenSSL.fips_mode == true
+      OpenSSL.fips_mode = false
+      OpenSSL.fips_mode == false
+    end
```

Comment:
This is not testing. Also, the mode can not necessarily be switched at runtime. I think all we can do here is:

    assert OpenSSL.fips_mode == true || OpenSSL.fips_mode == false,
           ".fips_mode returns a boolean value"